nwastdlib/vlans.py

```diff
@@ -252,36 +250,37 @@ def is_single_vlan(self) -> bool:
         except ValueError:
             return False
 
-    @classmethod
-    def __get_pydantic_core_schema__(cls, _source_type: Any, _handler: GetCoreSchemaHandler) -> CoreSchema:
-        return core_schema.no_info_after_validator_function(
-            cls._validate,
-            core_schema.str_schema(),
-            serialization=core_schema.plain_serializer_function_ser_schema(
-                cls._serialize,
-                info_arg=False,
-                return_schema=core_schema.str_schema(),
-            ),
-        )
-
-    @classmethod
-    def __get_pydantic_json_schema__(cls, core_schema_: CoreSchema, handler: GetJsonSchemaHandler) -> JsonSchemaValue:
-        json_schema = handler(core_schema_)
-        json_schema_resolved = handler.resolve_ref_schema(json_schema)
-        schema_override = {
-            "type": "string",
-            "format": "vlanrange",
-            "pattern": "^([1-4][0-9]{0,3}(-[1-4][0-9]{0,3})?,?)+$",
-            "examples": ["345", "20-23,45,50-100"],
-        }
-        return json_schema_resolved | schema_override
-
-    @staticmethod
-    def _validate(input_value: Union[str, VlanRanges]) -> VlanRanges:
-        if isinstance(input_value, VlanRanges):
-            return input_value
-        return VlanRanges(input_value)
-
-    @staticmethod
-    def _serialize(value: VlanRanges) -> str:
-        return str(value)
+
+def _serialize_vlanranges(value: VlanRanges) -> str:
+    return str(value)
+
+
+def _validate_vlanranges(input_value: Any) -> VlanRanges:
+    if isinstance(input_value, VlanRanges):
+        return input_value
+    return VlanRanges(input_value)
+
+
+def vlan_ranges_validator(json_schema_extra: dict | None = None) -> Any:
+    """Create a VlanRanges type annotated with validation.
+
+    Args:
+        json_schema_extra: schema to add to the standard vlanranges validator schema
+    """
+    json_schema_extra = {
+        "type": "string",
+        "format": "vlanrange",
+        "pattern": "^([1-4][0-9]{0,3}(-[1-4][0-9]{0,3})?,?)+$",
+        "examples": ["345", "20-23,45,50-100"],
+    } | (json_schema_extra or {})
+
+    return Annotated[
+        Any,
+        VlanRanges,
+        AfterValidator(_validate_vlanranges),
+        PlainSerializer(_serialize_vlanranges, when_used="json"),
```

Comment:
Just PlainSerializer(str, when_used="json") would also work

Reply:
It doesn't :)